File: esop2020/benchmarks/seahorn/4-just-rec/just-rec-00-0.c
#include "seahorn/seahorn.h"

extern int nd();

void just_rec(int *pa) {
  if (nd()) return;
  int b = nd();
  just_rec(&b);
}

int main() {
  int a = nd();
  int old_a = a;
  just_rec(&a);
  sassert(old_a == a);
}

File: esop2020/benchmarks/rust-horn/4-just-rec/just-rec-00-0.rs
pub fn rand<T>() -> T { unimplemented!() }

fn just_rec(ma: &mut i32) {
  if rand() {
    return;
  }
  let mut b = rand();
  just_rec(&mut b);
}
fn main() {
  let mut a = rand();
  let old_a = a;
  just_rec(&mut a);
  assert!(old_a == a);
}
